cfgov/v1/forms.py

```diff
@@ -54,42 +59,79 @@ def clean(self):
             raise forms.ValidationError(ERROR_MESSAGES['DATE_ERRORS']['one_required'])
         return cleaned_data
 
-class EventsFilterForm(forms.Form):
-    tags_select_attrs = {
-        'class': 'chosen-select',
-        'multiple': 'multiple',
-        'data-placeholder': 'Search for topics',
-    }
-    from_select_attrs = {
-        'class': 'js-filter_range-date js-filter_range-date__gte',
-        'type': 'text',
-        'placeholder': 'YYYY-MM',
-    }
-    to_select_attrs = from_select_attrs.copy()
-    to_select_attrs.update({
-        'class': 'js-filter_range-date js-filter_range-date__lte',
-    })
-
-    from_date = forms.DateField(required=False, input_formats=['%Y-%m-%d'],
-                                widget=DateInput(attrs=from_select_attrs))
-    to_date = forms.DateField(required=False, input_formats=['%Y-%m-%d'],
-                              widget=DateInput(attrs=to_select_attrs))
-    topics = forms.MultipleChoiceField(required=False, choices=[],
-                                       widget=SelectMultiple(attrs=tags_select_attrs))
-    @property
-    def field_queries(self):
-        return zip(['start_dt__gte', 'end_dt__lte', 'tags__name__in'],
-                   ['from_date', 'to_date', 'topics'])
+
+class FilterForm(forms.Form):
+    from_date = forms.DateField(
+        required=False,
+        input_formats=['%Y-%m-%d'],
+        widget=widgets.DateInput())
+    to_date = forms.DateField(
+        required=False,
+        input_formats=['%Y-%m-%d'],
+        widget=widgets.DateInput())
+    categories = forms.MultipleChoiceField(
+        required=False,
+        choices=ref.categories,
+        widget=widgets.CheckboxSelectMultiple())
+    topics = forms.MultipleChoiceField(
+        required=False,
+        choices=[],
+        widget=widgets.SelectMultiple())
+    authors = forms.MultipleChoiceField(
+        required=False,
+        choices=[],
+        widget=widgets.SelectMultiple())
 
     def __init__(self, *args, **kwargs):
-        super(EventsFilterForm, self).__init__(*args, **kwargs)
+        super(FilterForm, self).__init__(*args, **kwargs)
 
-        if 'topics' in self.fields:
-            options = list(set([tag for tags in [event.tags.names() for event
-                                                 in EventPage.objects.live()]
-                                for tag in tags]))
-            most = [(option, option) for option in options[:3]]
-            other = [(option, option) for option in options[3:]]
+        # Populate Topics' choices
+        topics_options = list(set([tag for tags in [page.tags.names() for page
+                                   in CFGOVPage.objects.live()]
+                                   for tag in tags]))
+        most = [(option, option) for option in topics_options[:3]]
```

Comment:
hmm so i'm just guess so let me know if this is incorrect. but it looks like ES used to keep tags ordered in most frequently used but i don't think this is the case for wagtail. So most frequently used tags logic would need some tweaking (maybe at a future date)